commit ac2167d75863000d7414ed7e5abfcad003f972f6
Author: Dominique Leuenberger <dimstar@opensuse.org>
Date:   Mon Aug 11 14:02:32 2025 +0200

    build: always use dummy.c to build a header-only dummy library
    
    Starting with Meson 1.9.0rc2, there is an error
    ERROR: Build target noinst has no sources
    when there are no sources assigned.

diff --git a/headers/dummy-msvc.c b/headers/dummy.c
similarity index 100%
rename from headers/dummy-msvc.c
rename to headers/dummy.c
diff --git a/headers/meson.build b/headers/meson.build
index 2aed919..46f746c 100644
--- a/headers/meson.build
+++ b/headers/meson.build
@@ -22,11 +22,9 @@ enums_xml = custom_target(
 if get_option('introspection')
   gir_sources = headers
 
-  # Use a dummy .c source for Visual Studio builds-it is not enough
-  # to send in just a header file to build a dummy library on MSVC
-  if cc.get_id() == 'msvc'
-    gir_sources += ['dummy-msvc.c']
-  endif
+  # Use a dummy .c source; Visual Studio and meson 1.9.0 do not accept
+  # header-only sources to build a dummy library
+  gir_sources += ['dummy.c']
 
   noinst_lib = shared_library('noinst',
                               gir_sources,
